Lib/ftplib.py

```diff
@@ -324,11 +324,14 @@ def makeport(self):
     def makepasv(self):
         """Internal: Does the PASV or EPSV handshake -> (address, port)"""
         if self.af == socket.AF_INET:
-            untrusted_host, port = parse227(self.sendcmd('PASV'))
-            if self.trust_server_pasv_ipv4_address:
-                host = untrusted_host
-            else:
-                host = self.sock.getpeername()[0]
+            try:
+                untrusted_host, port = parse227(self.sendcmd('PASV'))
+                if self.trust_server_pasv_ipv4_address:
+                    host = untrusted_host
+                else:
+                    host = self.sock.getpeername()[0]
+            except:
```

Comment:
Is it possible to have a less broad try-except? For instance, we only want to retry if `parse227(self.sendcmd('PASV'))` fails, not if `self.sock.getpeername()[0] fails.

Reply:
(and we also don't want to catch all possible exceptions)